server/routes/pages.js

```diff
@@ -38,7 +38,23 @@ async function handlePage(req, res) {
         if (exactMatches.length === 1) return res.redirect(exactMatches[0].path)
       }
 
-      res.render(template, {q, results, template: stringTemplate})
+      res.format({
+        html: () => {
+          res.render(template, {q, results, template: stringTemplate})
+        },
+
+        json: () => {
+          res.json(results.map((result) => ({
+            url: result.path,
+            title: result.prettyName,
+            lastUpdatedBy: (result.lastModifyingUser || {}).displayName,
+            modifiedAt: result.modifiedTime,
+            createdAt: result.createdTime,
+            id: result.id,
+            resourceType: result.resourceType
```

Comment:
Is there something equivalent for resourceType I could be returning for the categories and index endpoints to make the shape the same?

Reply:
Hey @tilgovi,

I don't have strong feelings about the shapes needing to match exactly, but if you'd like it looks like the same `resourceType` property should be available in the categories/index view.

The full set of properties that are available there are: 
```json
[
  "nodeType",
  "prettyName",
  "home",
  "homePrettyName",
  "id",
  "breadcrumb",
  "sort",
  "children",
  "path",
  "name",
  "mimeType",
  "parents",
  "webViewLink",
  "createdTime",
  "modifiedTime",
  "lastModifyingUser",
  "tags",
  "resourceType",
  "slug",
  "isTrashCan",
  "folder",
  "topLevelFolder"
  ]
```

Reply:
Looks like @SMores took care of adding `resourceType`. We can always add more later, but I feel good about starting with the set we've got now.